libos/src/sys/libos_flock.c

```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: LGPL-3.0-or-later */
+/*
+ * Implementation of system call "flock".
+ */
+#include <errno.h>
+
+#include "libos_fs.h"
+#include "libos_handle.h"
+#include "libos_internal.h"
+#include "libos_lock.h"
+#include "libos_process.h"
+#include "libos_table.h"
+#include "libos_thread.h"
+
+long libos_syscall_flock(int fd, int operation) {
+    int ret;
+
+    struct libos_handle_map* handle_map = get_thread_handle_map(NULL);
+    assert(handle_map);
+
+    struct libos_handle* hdl = get_fd_handle(fd, NULL, handle_map);
+    if (!hdl)
+        return -EBADF;
+
+    if(!hdl->dentry || !hdl->fs || !hdl->fs->fs_ops || !hdl->fs->fs_ops->flock) {
+        ret = -EINVAL;
+        goto out;
+    }
+
+    ret = hdl->fs->fs_ops->flock(hdl, operation);
```

Comment:
Wanted to finish a similar PR next week. Based on my understanding this is missing various semantics of `flock(...)`. I have opted in my implementation to map `flock(...)` to `fcntl(...)`. This also allows for NFS support and should avoid making any calls out of the enclave. Maybe missing something since this is not my forte.

Reply:
`fcntl` and `flock` use different file locks in Linux or it will cause deadlock.